src/DynamoCoreWpf/ViewModels/Preview/InfoBubbleViewModel.cs

```diff
@@ -50,56 +54,157 @@ public enum State
+        
+        // Relates to whether the info/warning/error message bodies display a button saying
+        // 'show more' or 'show less'.
+        private bool nodeInfoShowLessMessageVisible;
+        private bool nodeWarningsShowLessMessageVisible;
+        private bool nodeErrorsShowLessMessageVisible;
+
+        /// <summary>
+        /// Determines whether the show more/show less buttons are visible to the user at the Info Message level
+        /// </summary>
+        public bool NodeInfoShowMoreButtonVisible => NodeInfoSectionExpanded && NodeInfoIteratorVisible;
+
+        /// <summary>
+        /// Determines whether the show more/show less buttons are visible to the user at the Warning level
+        /// </summary>
+        public bool NodeWarningsShowMoreButtonVisible => NodeWarningsSectionExpanded && NodeWarningsIteratorVisible;
+        
+        /// <summary>
+        /// Determines whether the show more/show less buttons are visible to the user at the Error level
+        /// </summary>
+        public bool NodeErrorsShowMoreButtonVisible => NodeErrorsSectionExpanded && NodeErrorsIteratorVisible;
+
+        /// <summary>
+        /// Determines what the show more/show less buttons say at the Info Message level, which depends on whether
+        /// the NodeMessageVisibility at this level is set to CollapseMessages or ShowAllMessages.
+        /// </summary>
+        public bool NodeInfoShowLessMessageVisible
+        {
+            get => nodeInfoShowLessMessageVisible;
+            set
+            {
+                nodeInfoShowLessMessageVisible = value;
+                RaisePropertyChanged(nameof(NodeInfoShowLessMessageVisible));
+            }
+        }
+
+        /// <summary>
+        /// Determines what the show more/show less buttons say at the Warning level, which depends on whether
+        /// the NodeMessageVisibility at this level is set to CollapseMessages or ShowAllMessages.
+        /// </summary>
+        public bool NodeWarningsShowLessMessageVisible
+        {
+            get => nodeWarningsShowLessMessageVisible;
+            set
+            {
+                nodeWarningsShowLessMessageVisible = value;
+                RaisePropertyChanged(nameof(NodeWarningsShowLessMessageVisible));
+            }
+        }
+
+        /// <summary>
+        /// Determines what the show more/show less buttons say at the Error Message level, which depends on whether
+        /// the NodeMessageVisibility at this level is set to CollapseMessages or ShowAllMessages.
+        /// </summary>
+        public bool NodeErrorsShowLessMessageVisible
+        {
+            get => nodeErrorsShowLessMessageVisible;
+            set
+            {
+                nodeErrorsShowLessMessageVisible = value;
+                RaisePropertyChanged(nameof(NodeErrorsShowLessMessageVisible));
+            }
+        }
+
+        public string Content
```

Comment:
Let's add comments for this

Reply:
This is a property from the old code. There's a lot of obsolete code in the ViewModel now - I'll flag it as such.